File: Programming101/week0/prime_number_of_divisors.py
from is_prime import is_prime


def number_of_divisors(n):
    count = 0
    for number in range(1, n+1):
        if n % number == 0:
            count += 1
    return count


def prime_number_of_divisors(n):
    return is_prime(number_of_divisors(n))


def main():
    print(prime_number_of_divisors(7))
    print(prime_number_of_divisors(8))
    print(prime_number_of_divisors(9))

if __name__ == '__main__':
    main()

File: FrontendJavaScript/week0/2-A-Set-Of-Problems/set1/primeNumberOfDivisors.js
'use strict';

var primeNumberOfDivisors = function(n) {
  var isPrime = require('./isPrime').isPrime;
  var number = 0;
  var i = n;
  for(i; i>0; i--) {
    if (n % i === 0) {
      number += 1;
    }
  }
  return isPrime(number);
};

console.log(primeNumberOfDivisors(7) === true);
console.log(primeNumberOfDivisors(8) === false);
console.log(primeNumberOfDivisors(9) === true);
